src/amplitude-client.js

```diff
@@ -960,6 +994,27 @@ AmplitudeClient.prototype.setDeviceId = function setDeviceId(deviceId) {
   }
 };
 
+/**
+ * Sets the network transport type for events. Typically used to set to 'beacon'
+ * on an end-of-lifecycle event handler such as `onpagehide` or `onvisibilitychange`
+ * @public
+ * @param {string} transport - transport mechanism to use for events. Must be one of `http` or `beacon`.
+ * @example amplitudeClient.setDeviceId('45f0954f-eb79-4463-ac8a-233a6f45a8f0');
+ */
+AmplitudeClient.prototype.setTransport = function setTransport(transport) {
+  if (this._shouldDeferCall()) {
+    return this._q.push(['setTransport'].concat(Array.prototype.slice.call(arguments, 0)));
+  }
+
+  const t = transport.toLowerCase();
```

Comment:
not sure how to manage this in `validateTransport` but you should probably check that this is a string before using `toLowerCase`

Reply:
perhaps like `validateProperties` we coudl have `validateTransport` return a sanitary version of the string? :)